sparse/coo.py

```diff
@@ -2149,101 +2117,153 @@ def _elemwise_binary(func, self, other, *args, **kwargs):
     return COO(coords, data, shape=result_shape, has_duplicates=False)
 
 
-def _elemwise_binary_self_dense(func, self, other, *args, **kwargs):
-    assert isinstance(self, np.ndarray)
-    assert isinstance(other, COO)
+def _match_coo(*args):
+    """
+    Matches the coordinates for any number of input :obj:`COO` arrays.
+    Equivalent to "sparse" broadcasting for all arrays.
 
-    result_shape = _get_broadcast_shape(self.shape, other.shape)
+    Parameters
+    ----------
+    args : tuple[COO]
```

Comment:
If you're trying for parametrized python type annotations then I think it's supposed to be standard to use capitalized types like `List[COO]` or `Tuple[np.ndarray]`

Reply:
I don't know though, this is somewhat new to me.

Reply:
`tuple` and `list` tend to work better with intersphinx and code type annotations, so I tend to prefer those. Of course, I could import in something, but then that gives me PEP8 failures as I don't use it in code, just in docstrings.